nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/JoltTransformJSON.java

```diff
@@ -266,7 +294,7 @@ public void onTrigger(final ProcessContext context, ProcessSession session) thro
         try (final InputStream in = session.read(original)) {
             inputJson = JsonUtils.jsonToObject(in);
```

Comment:
This is where I'm wondering inputCharset was supposed to be applied for incoming json data (@mattyb149  any thoughts on this?)?  If so I think you may need a bit more manipulation to apply inputChartset before passing to jsonToObject (since it may not support passing inputstream with charset).

Reply:
Nope, that library supports two methods there - String JSON with String Charset name and InputStream.

Reply:
@travisneeley I had a chat with @mattyb149 offline and he confirmed the intent was to apply changes for incoming json.  Given JsonUtils.jsonToObject method doesn't support passing both InputStream and encoding or even allow a reader, it may be reasonable to use a ReaderInputStream (see https://commons.apache.org/proper/commons-io/javadocs/api-release/org/apache/commons/io/input/ReaderInputStream.html).

It does makes sense to also apply to the specification, since transformations can potentially add characters to the result or to simply match content. The same property for the input charset can be applied to both (we can't see a case where a separate charset would be needed for the spec).